### PR TITLE
Sphinx documentation for 0.8

### PR DESCRIPTION
**Pull request by MateuszKubuszok**
TODO:
- [x] change Gitter links to GitHub Discussions
- [x] finish "Migration to 0.8" section
- [x] finish "Libraries with smart constructors" section
- [x] finish "Protocol Buffers" section
- [x] finish "Debugging macros" section
  - [x] link to this section in scaladocs
- [x] finish "Known issues and limitations" section
- [x] wire new sections in ToC

### CODE REVIEW

File: docs/source/migration/migration-to-0.8.rst [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  implicit was generated by automatic derivation. If it came from
+  ``Transformer.derive`` or ``PartialTransformer.derive`` it was discarded and
+  macro attempted to derive it again without wrapping result in a type class.
+
+  It both complicated code as well as increased compilation times as each field
+  or subtype would attempt to summon implicit and then discard it if it didn't
+  came from user. Splitting types allows compiler to not summon any implicit if
+  user haven't provided any.
+
+  The consequence is only visible if there is some ``implicit def`` which takes
```
Comment (krzemin) [inline docs/source/migration/migration-to-0.8.rst:29]: This paragraph is quite hard to comprehend. Maybe some short example would do the job much better?

File: docs/source/migration/migration-to-0.8.rst [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+- lifted transformers, deprecated in 0.7.0, got removed in favor of partial
+  transformers. Since 0.7.0 migration guide from lifted to partial is available
+  in ``partial transformers`` migrating from :ref:`lifted section<migrating-from-lifted>`.
+- ``.enableUnsafeOption`` option was removed - if ``Option`` unwrapping is
+  needed, it is recommended to use :ref:`partial transformers<partial-transformers>`
+- types returned by automatic derivation got split from types that are used
+  for user-provided transformations and configured (semiautomatic) derivation:
+  ``Transformer`` got split into ``Transformer`` and ``Transformer.AutoDerived``
+  while ``PartialTransformer`` got split into ``PartialTransformer`` and
+  ``PartialTransformer.Autoderived``.
```
Comment (krzemin) [inline docs/source/migration/migration-to-0.8.rst:16]: AutoDerived

File: docs/source/cookbook/libraries-with-smart-constructors.rst [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+Any type which uses a smart constructor (returning parsed result rather than
+throwing an exception) would require partial transformer rather than total
+transformer to convert.
+
+If there is no common interface which could be summoned as implicit for
+performing smart construction:
+
+.. code-block:: scala
+
+  // assuming Scala 3 or -Xsource:3 for fixed private constructors
```
Comment (krzemin) [inline docs/source/cookbook/libraries-with-smart-constructors.rst:13]: why this assumption is relevant?

**Review by krzemin — DISMISSED** [review]

File: docs/source/cookbook/libraries-with-smart-constructors.rst (hunk 3 above)
Comment (MateuszKubuszok) [reply docs/source/cookbook/libraries-with-smart-constructors.rst:13]: Without it `private` doesn't prevent generation of `Username.apply` nor `.copy()` so it is not a class with a smart constructor

File: docs/source/migration/migration-to-0.8.rst (hunk 2 above)
Comment (MateuszKubuszok) [reply docs/source/migration/migration-to-0.8.rst]: Done

File: docs/source/migration/migration-to-0.8.rst (hunk 1 above)
Comment (MateuszKubuszok) [reply docs/source/migration/migration-to-0.8.rst:29]: Rewritten

**Pull request merged**